test/e2e/data/user.go

```diff
@@ -0,0 +1,79 @@
+package data
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common"
+)
+
+const (
+	// build-in dbroles
+	RoleBuildInAdmin        string = "atlasAdmin"
+	RoleBuildInReadWriteAny string = "readWriteAnyDatabase"
+	RoleBuildInReadAny      string = "readAnyDatabase"
+
+	DefaultDatabaseName = "admin"
+)
+
+func BasicUser(crName, atlasUserName string, add ...func(user *v1.AtlasDatabaseUser)) *v1.AtlasDatabaseUser {
+	user := &v1.AtlasDatabaseUser{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crName,
+		},
+		Spec: v1.AtlasDatabaseUserSpec{
+			Project: common.ResourceRefNamespaced{
+				Name: ProjectName,
+			},
+			Username: atlasUserName,
+		},
+	}
+	for _, f := range add {
+		f(user)
+	}
+	return user
+}
+
+func WithSecretRef(name string) func(user *v1.AtlasDatabaseUser) {
```

Comment:
One might think these should be methods `func (u *v1.AtlasDatabaseUser) With...()` 🤔

Reply:
Just so you can chain them like `user.WithAdminRight().WithCredit().WithSuperPowers()` =)

Reply:
Now it can be done like this:
`data.BasicUser("admin", "user1", data.WithSecretRef("dbuser-secret-u1"), data.WithAdminRole())`

Reply:
Well, I hardly think your example is an argument for this change but it's probably just a matter of preference so feel free to leave it as is)